### ISSUE TITLE
[sass|scss] ::slotted() selector throws fatal error

### ISSUE THREAD

**Issue by DizWARE** [sasstools/sass-lint#1053]
I'm currently in the process of upgrading a project to use W3C spec Custom Components, and am running into an issue with the `::slotted` psuedo-element. 

Here's some links for reference:
[W3C Spec for Shadow DOM](https://w3c.github.io/webcomponents/spec/shadow/)
[W3C Spec for ::slotted](https://drafts.csswg.org/css-scoping/#slotted-pseudo)
[Custom Elements - MDN](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Custom_Elements)

**What version of Sass Lint are you using?**
1.10.2

**Please include any relevant parts of your configuration**
As far as I can tell this is not a configuration issue. My guess, if I had one on what is happening, is that this is the first pseudo-element that takes parameters, and the linting engine is getting confused by the parentheses at the end of the selector i.e. `::slotted(.my-class)` 

**What did you do? Please include the actual source code causing the issue.**
Here's an example of the issue. 
The HTML would be:
```HTML
<!--In the shadow DOM-->
<slot name="widget-icon"></slot>
```

Here's the sass. Essentially, the css will style the top-level element/s with the slot name "widget-icon"; Any selector can be used in place of the '*'.
```SASS
slot[name=widget-icon]::slotted(*) {
    ...
}
```

This could also be written as the follow:

```SASS
slot {
    &[name=widget-icon]::slotted(*) {
            ...
     }
}
```
-or-
```SASS
slot {
    &[name=widget-icon] {
        &::slotted(*) {
            ...
        }
    }
}
```

**What did you expect to happen?**
This shouldn't produce any linting errors as long as the syntax is correct, since it is valid way of writing the selector for a slotted item.

**What actually happened? Please include any error messages given to you by Sass Lint.**
Every use of the `::slotted` pseudo-element produces this error: 
```diff
- Please check validity of the block starting from line #X  Fatal
```
Building and running the compiled css code works as expected, so this is strictly an linting bug.

**If you're using a IDE plugin have you tried the CLI too?**
I have tried using gulp-sass-lint and the sass-lint CLI; Not actually using any integration with my IDE.

**Comment by DanPurdy** [comment]
Hi there, you'll need to report this to the AST we use, gonzales-pe the fatal errors are thrown by that.

**Comment by DizWARE** [comment]
Ok thanks. I'll go ahead and do that.

**Comment by bgriffith** [comment]
`::slotted` is now supported so these issues should be fixed as of `v1.11.1`
